packages/legend-studio/src/stores/ViewerStore.ts

```diff
@@ -450,47 +375,131 @@ export class ViewerStore {
+          entities,
+          this.editorStore.graphManagerState.graphBuildState,
+        )) as GraphBuilderReport;
+
+      // report
+      stopWatch.record(GRAPH_MANAGER_EVENT.GRAPH_INITIALIZED);
+      const graphBuilderReportData = {
+        timings: {
+          [GRAPH_MANAGER_EVENT.GRAPH_INITIALIZED]: stopWatch.getRecord(
+            GRAPH_MANAGER_EVENT.GRAPH_INITIALIZED,
+          ),
+        },
+        dependencies: dependency_buildReport,
+        graph: graph_buildReport,
+      };
+      this.editorStore.applicationStore.log.info(
+        LogEvent.create(GRAPH_MANAGER_EVENT.GRAPH_INITIALIZED),
+        graphBuilderReportData,
+      );
+      GraphManagerTelemetry.logEvent_GraphInitialized(
+        this.editorStore.applicationStore.telemetryService,
+        graphBuilderReportData,
+      );
+
+      // fetch available file generation descriptions
+      yield flowResult(
+        this.editorStore.graphState.graphGenerationState.fetchAvailableFileGenerationDescriptions(),
+      );
+      yield flowResult(
+        this.editorStore.graphState.graphGenerationState.externalFormatState.fetchExternalFormatsDescriptions(),
+      );
+
+      return true;
+    } catch (error) {
+      assertErrorThrown(error);
+
+      // if graph builder fails, we fall back to text-mode
+      this.editorStore.applicationStore.log.error(
+        LogEvent.create(GRAPH_MANAGER_EVENT.GRAPH_BUILDER_FAILURE),
+        error,
+      );
+      if (error instanceof DependencyGraphBuilderError) {
+        // no recovery if dependency models cannot be built, this makes assumption that all dependencies models are compiled successfully
+        // TODO: we might want to handle this more gracefully when we can show people the dependency model element in the future
+        this.editorStore.applicationStore.notifyError(
+          `Can't initialize dependency models. Error: ${error.message}`,
+        );
+      } else if (error instanceof GraphDataDeserializationError) {
+        // if something goes wrong with de-serialization, we can't really do anything but to alert
+        this.editorStore.applicationStore.notifyError(
+          `Can't deserialize graph. Error: ${error.message}`,
+        );
+      } else if (error instanceof GraphBuilderError) {
+        // TODO: we should split this into 2 notifications when we support multiple notifications
+        this.editorStore.applicationStore.notifyError(
+          `Can't build graph. Redirected to text mode for debugging. Error: ${error.message}`,
+        );
+        try {
+          const editorGrammar =
+            (yield this.editorStore.graphManagerState.graphManager.entitiesToPureCode(
+              entities,
+            )) as string;
+          yield flowResult(
+            this.editorStore.grammarTextEditorState.setGraphGrammarText(
+              editorGrammar,
+            ),
+          );
+        } catch (error2) {
+          assertErrorThrown(error2);
```

Comment:
should we continue if it fails to convert to pure text. 
Almost falls into a similar category of `if something goes wrong with de-serialization, we can't really do anything but to alert`.

Reply:
Nice catch! Updated